# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1459977061 14400
#      Wed Apr 06 17:11:01 2016 -0400
# Node ID fe5ed6d7fb10487b9c0f3bc36162f280fbe7561a
# Parent  ec77a07e422049a121656be60253a015ca62a61b
* dlmread.cc (Fdlmread): Search loadpath for file (bug #47628).

diff --git a/libinterp/corefcn/dlmread.cc b/libinterp/corefcn/dlmread.cc
--- a/libinterp/corefcn/dlmread.cc
+++ b/libinterp/corefcn/dlmread.cc
@@ -210,16 +210,18 @@ such as text, are also replaced by the @
 
   if (args(0).is_string ())
     {
       // Filename.
       std::string fname (args(0).string_value ());
 
       std::string tname = file_ops::tilde_expand (fname);
 
+      tname = find_data_file_in_load_path ("dlmread", fname);
+
       input_file.open (tname.c_str (), std::ios::in);
 
       if (! input_file)
         error ("dlmread: unable to open file '%s'", fname.c_str ());
 
       input = &input_file;
     }
   else if (args(0).is_scalar_type ())
